### PR TITLE
feat: update connectors table and load connector logic

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

1. update `connectors` table
2. add alteration script
3. update load connector logic

### CODE REVIEW

File: packages/core/src/connectors/index.ts [hunk 1]
```diff
@@ -92,23 +92,27 @@ const loadConnectors = async () => {
...
-  const connectorMap = new Map(connectors.map((connector) => [connector.id, connector]));
 
-  const logtoConnectors = await loadConnectors();
+  const virtualConnectors = await loadConnectors();
 
-  return logtoConnectors.map((element) => {
-    const { id } = element.metadata;
-    const connector = connectorMap.get(id);
+  return Promise.all(
+    connectors.map(async (connector) => {
```
Comment (wangsijie) [inline packages/core/src/connectors/index.ts:99]: That is a mapping by sync function? And the `Promise.all` is useless

File: packages/console/src/pages/ConnectorDetails/components/ConnectorContent.tsx [hunk 2]
```diff
@@ -55,11 +55,11 @@ const ConnectorContent = ({ isDeleted, connectorData, onConnectorUpdated }: Prop
...
     }
 
-    const { metadata, ...rest } = await api
+    const connectorResponse = await api
       .patch(`/api/connectors/${connectorData.id}`, { json: { config: result.data } })
-      .json<Connector & { metadata: ConnectorMetadata; type: ConnectorType }>();
+      .json<ConnectorResponse>();
 
-    onConnectorUpdated({ ...rest, ...metadata });
+    onConnectorUpdated(connectorResponse);
```
Comment (wangsijie) [inline packages/console/src/pages/ConnectorDetails/components/ConnectorContent.tsx:62]: Let's focus on the target, leave the data structure as it is before.

File: packages/schemas/alterations/next-1668666590-support-multiple-connector-instances.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { sql } from 'slonik';
+
+import type { AlterationScript } from '../lib/types/alteration';
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    await pool.query(sql`
+      DELETE FROM connectors WHERE enabled = false;
```
Comment (wangsijie) [inline packages/schemas/alterations/next-1668666590-support-multiple-connector-instances.ts:8]: Actually we don't have to delete them?

**Review by wangsijie — CHANGES_REQUESTED** [review]

File: packages/core/src/connectors/index.ts (hunk 1 above)
Comment (darcyYe) [reply packages/core/src/connectors/index.ts]: Fixed.

File: packages/core/src/connectors/index.ts [hunk 4]
```diff
@@ -92,20 +92,22 @@ const loadConnectors = async () => {
...
 
-  const logtoConnectors = await loadConnectors();
+  const virtualConnectors = await loadConnectors();
 
-  return logtoConnectors.map((element) => {
-    const { id } = element.metadata;
-    const connector = connectorMap.get(id);
+  return connectors.map((connector) => {
+    const { metadata, connectorId } = connector;
+    const pickedVirtualConnector = virtualConnectors.find(
```
Comment (wangsijie) [inline packages/core/src/connectors/index.ts:100]: nit: just `virtualConnector`

**Review by wangsijie — APPROVED** [review]

**Pull request merged**